components/calendar/src/helpers.rs

```diff
@@ -13,6 +13,17 @@ pub fn div_rem_euclid(n: i32, d: i32) -> (i32, i32) {
     }
 }
 
+/// [`div_rem_euclid`] for 64-bit inputs
+pub fn div_rem_euclid64(n: i64, d: i64) -> (i64, i64) {
+    debug_assert!(d > 0);
+    let (a, b) = (n / d, n % d);
+    if n >= 0 || b == 0 {
+        (a, b)
+    } else {
+        (a - 1, d + b)
+    }
```

Comment:
Isn't this `(n.div_euclid(d), n.rem_euclid(d))`?

Reply:
Good observation, but this is pre-existing code. Added comments and a test.